src/mainwindow.cpp

```diff
@@ -1265,22 +1265,36 @@ void MainWindow::FindImpl(int offset, bool findHighlight)
         else if (i < 0)
             i = model->rowCount() - 1;
 
-        for (SearchOpt searchOpt : filters)
+        // If highlight only mode is on, don't search hidden rows
+        bool searchRow = true;
+        if (model->m_highlightOnlyMode)
         {
-            for (COL col : searchOpt.m_keys)
+            QModelIndex idx;
+            if(tree->isRowHidden(i, idx))
             {
-                QModelIndex idx = model->index(i, col);
-                QString data = (col == COL::Value) ?
-                    model->GetValueFullString(idx, true) :
-                    model->data(idx, Qt::DisplayRole).toString();
-                if (searchOpt.HasMatch(data))
+                searchRow = false;
+            }
+        }
+
+        if (searchRow)
```

Comment:
I think we don't need to check m_highlightOnlyMode, as we want to skip all hidden rows regardless of what mode.
Also could move the QModelndex outside of the block so we don't need to create and destroy one with each iteration.  So it could be just
```
    if  (!tree->isRowHidden(i, idx))
    { // search with filters }
```

Reply:
It's funny you bring up the m_highlightOnlyMode check. It turns out we only set rows to hidden with m_highlightOnlyMode anyway. The actual right click -> hide actually removes the rows instead of hiding them. But I can make this just check for hidden rows anyway, it might be better for future proofing in case other stuff starts using setRowHidden().